test/reviewController.test.js

```diff
@@ -58,70 +59,219 @@ describe('ReviewController API Tests', () => {
+
+      //create second review for the same album
+      const secondReviewData = {
+        rating: 4,
+        title: 'Loved this review',
+        comment: 'Recommend it"',
+        user: user._id,
+        album: album._id,
+      };
+      const res = await request(app)
+        .post(`/api/v1/reviews/album/${album._id}`)
+        .set('Cookie', signedCookie)
+        .send(secondReviewData);
+      // this user already submitted a review for this album
+      expect(res.status).toBe(StatusCodes.CONFLICT);
+    });
+
+    //when album was not found
+    it('should not create a review and return a 404 if the album is not found ', async () => {
+      const signedCookie = await loginAndReturnCookie(userCredentials);
+
+      album = await Album.findOneAndDelete({ albumName: 'Unique Album' });
+
+      const newReviewData = {
+        rating: 4,
+        title: 'Loved this review',
+        comment: 'Recommend it"',
+        user: user._id,
+        album: album._id,
+      };
+      const response = await request(app)
+        .post(`/api/v1/reviews/album/${album._id}`)
+        .set('Cookie', signedCookie)
+        .send(newReviewData);
 
-  //get all reviews for this particular album
-  it('should get reviews for a specific album', async () => {
-    // Try to find an album; if not found, create a default one
-    const album = await Album.findOne({ albumName: 'Unique Album' });
-    const response = await request(app).get(
-      `/api/v1/reviews/album/${album._id}`
-    );
-
-    expect(response.statusCode).toBe(200);
-    expect(response.body).toHaveProperty('allProductReviews');
-    expect(response.body.allProductReviews).toHaveLength(1);
-    expect(response.body).toHaveProperty('count');
-    expect(response.body.count).toBe(1);
-    // If no reviews are found, allProductReviews should be an empty array and count should be 0
-    if (response.body.count === 0) {
-      expect(response.body.allProductReviews).toEqual([]);
-    }
+      expect(response.status).toBe(StatusCodes.NOT_FOUND);
+    });
+  });
+  describe('test the update review endpoint', () => {
+    //  update a review
+    it('should update a review successfully', async () => {
+      const signedCookie = await loginAndReturnCookie(userCredentials);
+
+      const existingReview = await Review.findOne({});
+
+      const updatedReviewData = {
+        rating: 1,
+        title: 'Don not buy this album',
+        comment: 'Awful"',
+      };
+
+      // Check if the user is the author of the review
+      const authenticatedUserId = 'mockUserId'; // Replace with actual authenticated user's ID
+
+      if (existingReview.user.toString() !== authenticatedUserId) {
+        // User is not the author of the review, respond with unauthorized error
+        expect(StatusCodes.UNAUTHORIZED).toBe(StatusCodes.UNAUTHORIZED); // //AKOS: this one works but I feel it's wrong, couldn't write here 'respone', the terminal was complaining
+      }
```

Comment:
We can remove these lines; they're not checking anything within the test specifically; and I think you have already updated the code to test the case where a non-authorized user tries to update a review :+1:

Reply:
thank you, done!